pkg/crawler/crawler.go

```diff
@@ -5,31 +5,57 @@ import (
 	"encoding/hex"
 	"encoding/xml"
 	"fmt"
-	"github.com/aquasecurity/trivy-java-db/pkg/fileutil"
-	"github.com/aquasecurity/trivy-java-db/pkg/types"
+	"hash/fnv"
 	"io"
 	"log"
 	"net/http"
+	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 	"sync"
+	"time"
+
+	"github.com/aquasecurity/trivy-java-db/pkg/fileutil"
+	"github.com/aquasecurity/trivy-java-db/pkg/types"
+	"github.com/google/licenseclassifier/v2/tools/identify_license/backend"
+	"github.com/samber/lo"
 
 	"github.com/PuerkitoBio/goquery"
 	"github.com/hashicorp/go-retryablehttp"
+	cmap "github.com/orcaman/concurrent-map/v2"
+	"golang.org/x/net/html/charset"
 	"golang.org/x/sync/semaphore"
 	"golang.org/x/xerrors"
 )
 
 const mavenRepoURL = "https://repo.maven.apache.org/maven2/"
 
 type Crawler struct {
-	dir  string
-	http *retryablehttp.Client
+	dir        string
+	licensedir string
+	http       *retryablehttp.Client
 
 	rootUrl string
 	wg      sync.WaitGroup
 	urlCh   chan string
 	limit   *semaphore.Weighted
+	opt     Option
+
+	// license classifier
+	classifier *backend.ClassifierBackend
+
+	// list of temporary license files
+	files []string
+
+	// map of temporary license files created to license metadata
+	filesLicenseMap cmap.ConcurrentMap[string, License]
+
+	// processed temporary license files tracking
+	processedFileMap map[string]struct{}
```

Comment:
Can we remove `processedFileMap` and use only `filesLicenseMap`?
https://github.com/aquasecurity/trivy-java-db/blob/5ec049b1de6facc842d6a346fab72bb86c06df4b/pkg/crawler/crawler.go#L390-L402
```
				licenseVal, ok := c.filesLicenseMap.Get(r.Filename)
                                if !ok{

				// Pick results where confidence > 80%
				if r.Confidence > 0.8 {

					// update uniqueLicenseKeys
					licenseVal.NormalizedLicense = r.Name
					c.filesLicenseMap.Set(r.Filename, licenseVal)
				}
			}
```

Reply:
Yup. My bad. Went through a bunch of iterations. It's not useful anymore.  Let me get rid of it

Reply:
Oh wait.
So the intent is to skip processing for a file already processed.

License classifier returns multiple matches for a given license file/text with confidence values. Intend to pick the one with max confidence.

Reply:
` if !ok{` cannot be used since filesLicenseMap will always have default values set as part of https://github.com/aquasecurity/trivy-java-db/pull/21/files#diff-526a7142f7b15a72eab215dcfa540f5b6d502951a2f9f3b34dd900013e82c675R426

Reply:
One enhancement could be to have a key `LicenseClassificationDone bool` as part of the License struct being saved as value in filesLicenseMap. What do you think?

Reply:
If we save only unique URLs (as I wrote earlier) - we will not process duplicates.

Reply:
I think we can update `files` to use only uniq files for classifier. wdyt?

Reply:
> If we save only unique URLs (as I wrote earlier) - we will not process duplicates.

Assume i have license.txt file with some license text. When we send this to google license classifier, it can return more than 1 matches with same/different confidence values

M1,0.7
M2,0.9
M3,0.5
M4,0.85

All reference the same license file. I idea is to pick M2 only in the above example.

Since license classification is being done for multiple files at the same time, the results set can contain M*N results

i.e M files and N results per file

Reply:
Can we only get 1 result per file with maximum confidence values?

Reply:
Unfortunately no. Would have loved to fork the license classifier repo and add a bunch of enhancements to suit our use case, but that will require a decent amount of work.  For instance, i would have loved to handle the result aggregation better. Instead of list it should return a map of license file to list of results sorted by confidence.

Also an enhancement as you suggested to return only the result with maximum confidence.

We can also add an in house implementation of the classifier interface exposed by the library, but that will again be a decent amount of effort.

Reply:
They supported it earlier https://github.com/google/licenseclassifier/blob/c1ed8fcf4babdf7c37d872cf6da5b1c32907d34c/tools/identify_license/identify_license.go#L45, but not anymore in the latest version

Reply:
okay, i see 1 case:
we will add `confidence` field in license.
and create `readedFiles` (you can change name) map[string]license (file name -> license).

use next logic here (https://github.com/aquasecurity/trivy-java-db/blob/30bc298f3a944e4b0dfdf50ad1313dce0f71d2fa/pkg/crawler/crawler.go#L391-L409)
if lic, ok := readedFiles[fileName]; ok{
if lic.Confidence < result.Confidence{
save to readedFiles
}}

something like that. 
and we will init this map in `classifyLicense`.
wdyt?

Reply:
> if r.Confidence < 0.8 { 
 				break 
 			} 

I think we need to get better result.

Reply:
Let me check.

> I think we need to take better confidence.

Sure. I can increase it to 90% / 95%

Reply:
> Sure. I can increase it to 90% / 95%

Trivy uses 0.9 (`specify license classifier's confidence level (default 0.9)`). Let's use this value.

Reply:
Sounds good. Have got rid of the processedFileMap. Using the classification confidence as a field in the license struct as you suggested. Thanks for pointing it out. Hope its cleaner now